Sources/libhostmgr/libhostmgr.swift

```diff
@@ -0,0 +1,312 @@
+import Foundation
+import prlctl
+
+/// Downloads and registers an image by name
+///
+/// This method is the preferred way to install a remote image on a VM Host.
+public func fetchRemoteImage(name: String) async throws {
+    try await downloadRemoteImage(name: name)
+
+    Console.success("VM \(name) is ready")
+}
+
+/// Downloads a remote image with the given `name` and places it in the image storage directory
+///
+/// Does not import or register the image – this method only handles download.
+@discardableResult
+public func downloadRemoteImage(
+    name: String,
+    remoteRepository: RemoteVMRepository = RemoteVMRepository()
+) async throws -> URL {
+    guard let remoteImage = try await remoteRepository.getImage(named: name) else {
+        Console.crash(message: "Unable to find remote image: \(name)", reason: .unableToFindRemoteImage)
+    }
+
+    return try await downloadRemoteImage(remoteImage)
+}
+
+/// Downloads the given remote VM image to the given directory
+///
+/// Does not import or register the images – this method only handles download.
+@discardableResult
+public func downloadRemoteImage(
+    _ remoteImage: RemoteVMImage,
+    remoteRepository: RemoteVMRepository = RemoteVMRepository(),
+    storageDirectory: URL = Configuration.shared.vmStorageDirectory
+) async throws -> URL {
+
+    // If this is the first run, the storage directory may not exist, so we'll create it just in case
+    try FileManager.default.createDirectory(at: storageDirectory, withIntermediateDirectories: true)
+
+    let availableStorageSpace = try FileManager.default.availableStorageSpace(
+        forVolumeContainingDirectoryAt: storageDirectory
+    )
+
+    guard remoteImage.imageObject.size < availableStorageSpace else {
+        Console.crash(
+            message: [
+                "Unable to download \(remoteImage.fileName) (\(Format.fileBytes(remoteImage.imageObject.size)))",
+                "not enough local storage available (\(Format.fileBytes(availableStorageSpace)))"
+            ].joined(separator: " - "),
```

Comment:
Nitpick: we could use multiline string here instead of calling the join function?

Reply:
I couldn't think of an easy way to avoid ridiculous line lengths here so I did this (which obviously has a small runtime cost). If there's a better approach I'd be into it, but this was just quick and easy 😅